image/mkosi/README.md

```diff
@@ -0,0 +1,187 @@
+VMGS blobs for testing and release images already exist.
+
+First, create a disk without embedded MOK EFI variables.
+
+```sh
+# set these variables
+export AZURE_SECURITY_TYPE=ConfidentialVM # or TrustedLaunch
+export AZURE_RESOURCE_GROUP_NAME=
+
+export AZURE_REGION=northeurope
+export AZURE_DISK_NAME=constellation-$(date +%s)
+export AZURE_SNAPSHOT_NAME=${AZURE_DISK_NAME}
+export AZURE_RAW_IMAGE_PATH=${PWD}/mkosi.output.azure/fedora~36/image.raw
+export AZURE_IMAGE_PATH=${PWD}/mkosi.output.azure/fedora~36/image.vhd
+export AZURE_VMGS_FILENAME=${AZURE_SECURITY_TYPE}.vmgs
+export BLOBS_DIR=${PWD}/blobs
+upload/pack.sh azure "${AZURE_RAW_IMAGE_PATH}" "${AZURE_IMAGE_PATH}"
+upload/upload_azure.sh --disk-name "${AZURE_DISK_NAME}-setup-secure-boot" ""
+secure-boot/azure/launch.sh -n "${AZURE_DISK_NAME}-setup-secure-boot" -d --secure-boot true --disk-name "${AZURE_DISK_NAME}-setup-secure-boot"
+```
+
+Ignore the running launch script and connect to the serial console once available.
+The console shows the message "Verification failed: (0x1A) Security Violation". You can import the MOK certificate via the UEFI shell:
+
+Press OK, then ENTER, then "Enroll key from disk".
+Select the following key: `/EFI/loader/keys/auto/db.cer`.
+Press Continue, then choose "Yes" to the question "Enroll the key(s)?".
+Choose reboot.
+
+Extract the VMGS from the running VM (this includes the MOK EFI variables) and delete the VM:
+
+```sh
+secure-boot/azure/extract_vmgs.sh --name "${AZURE_DISK_NAME}-setup-secure-boot"
+secure-boot/azure/delete.sh --name "${AZURE_DISK_NAME}-setup-secure-boot"
+```
+
+</details>
+
+## Upload to CSP
+
+<details>
+<summary>GCP</summary>
+
+- Install `gcloud` and `gsutil` (see [here](https://cloud.google.com/sdk/docs/install))
+- Login to GCP (see [here](https://cloud.google.com/sdk/docs/authorizing))
+- Prepare secure boot PKI (see `secure-boot/genkeys.sh`)
+
+```sh
+# set these variables
+export GCP_IMAGE_FAMILY=
+export GCP_IMAGE_NAME=
+export PKI=${PWD}/pki
+
+export GCP_PROJECT=constellation-images
+export GCP_REGION=europe-west3
+export GCP_BUCKET=constellation-images
+export GCP_RAW_IMAGE_PATH=${PWD}/mkosi.output.gcp/fedora~36/image.raw
+export GCP_IMAGE_FILENAME=$(date +%s).tar.gz
+export GCP_IMAGE_PATH=${PWD}/mkosi.output.gcp/fedora~36/image.tar.gz
+upload/pack.sh gcp ${GCP_RAW_IMAGE_PATH} ${GCP_IMAGE_PATH}
+upload/upload_gcp.sh
+```
+
+</details>
+
+<details>
+<summary>Azure</summary>
+
+- Install `az` and `azcopy` (see [here](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli))
+- Login to Azure (see [here](https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli))
+- Prepare secure boot PKI (see `secure-boot/genkeys.sh`)
+- [Prepare virtual machine guest state (VMGS) with customized NVRAM or use existing VMGS blob](#azure-secure-boot)
+
+```sh
+# set these variables
+export AZURE_GALLERY_NAME=
+export AZURE_IMAGE_DEFINITION=
+export AZURE_IMAGE_VERSION=
+export AZURE_VMGS_PATH=
```

Comment:
Maybe add some placeholders for them to know what format is required here (e.g. just the gallery name, or the whole gallery URL (sub-)path?).